aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -349,13 +349,14 @@ impl<K: Debug + ModulePath, T: TransactionOutput, E: Debug + Send + Clone>
 
     // Must be executed after parallel execution is done, grabs outputs. Will panic if
     // other outstanding references to the recorded outputs exist.
-    pub(crate) fn take_output(&self, txn_idx: TxnIndex) -> ExecutionStatus<T, Error<E>> {
-        let owning_ptr = self.outputs[txn_idx as usize]
-            .swap(None)
-            .expect("[BlockSTM]: Output must be recorded after execution");
-
-        Arc::try_unwrap(owning_ptr)
-            .map(|output| output.output_status)
-            .expect("[BlockSTM]: Output should be uniquely owned after execution")
+    pub(crate) fn try_take_output(
```

Comment:
why do we need to change return type here?

Reply:
We need to change the type here as IIUC, the output status might be empty for cases where we have error or skip rest. Am I missing something?

Reply:
Oh, my bad, in my mind I was imagining doing it at commit (and then we shouldn't need the option, nor the index to reduce every time - we would just know how many got committed and fill it up)